doc/architecture/frontend.md

```diff
@@ -0,0 +1,57 @@
+# Frontend Architecture
+
+## Structure
+
+- `__mocks__/`: Contains mocked modules for tests
+- `assets/`: Contains asset files like images
+- `bindings/`: Contains the generated API bindings
+- `components/`: Contains components that are shared across features
+- `config/`: Contains initialization code for the environment and libraries
+- `features/`: Contains the distinct features of the application
+  - `dark_mode/`: Contains the dark_mode feature (Switcher and store)
+  - `imprint_page/`: Contains the imprint page
+  - `landing_page/`: Contains the landing page
+  - `map_planning/`: Contains the map with layers
+  - `seeds/`: Contains the seed routes
+- `generated/`: Contains generated code, for now only docs
+- `hooks/`: Contains hooks that are shared across features
+- `icons/`: Contains svg icons
+- `routes/`: Contains the routes of the application
+- `styles/`: Contains the global and miscellaneous style sheets
+- `utils/`: Contains additional utility functions
```

Comment:
Also here some guideline is needed. E.g. at least two components want to share the code or similar.